src/imagestyle/imagestyle.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module image/imagestyle/imagestyle
+ */
+
+import Plugin from 'ckeditor5-core/src/plugin';
+import ImageStyleEngine from './imagestyleengine';
+import ButtonView from 'ckeditor5-ui/src/button/buttonview';
+
+/**
+ * The image style plugin.
+ *
+ * Uses {@link module:image/imagestyle/imagestyleengine~ImageStyleEngine}.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class ImageStyle extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ ImageStyleEngine ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const styles = this.editor.config.get( 'image.styles' );
+
+		for ( let style of styles ) {
+			this._createButton( style );
+		}
+
+		// If there is no default image toolbar configuration, add all image styles buttons.
+		const imageToolbarConfig = this.editor.config.get( 'image.toolbar' );
```

Comment:
This is not going to work this way... Because, if there's any other plugin which would like to extend this toolbar, it won't do that, because the toolbar is already set by this plugin.

So, unfortunately, we must make this a bit more complex.

1. The assumption is that `ImageToolbar` is loaded and sets `image.defaultToolbar` to an empty array in its constructor (which is "before init"). If it's not loaded, the property doesn't exist.
2. If the property exists, the other plugins should then push their default buttons to this array in their `init()` callbacks.
3. The ImageToolbar should then use `image.toolbar` or `image.defaultToolbar`.

Sounds reasonable?